pkg/grpcfactory/client_test.go

```diff
@@ -0,0 +1,130 @@
+package grpcfactory_test
+
+import (
+	"context"
+	"net"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/scality/cosi-driver/pkg/grpcfactory"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	cosi "sigs.k8s.io/container-object-storage-interface-spec"
+)
+
+type MockIdentityServer struct {
+	cosi.UnimplementedIdentityServer
+}
+
+type MockProvisionerServer struct {
+	cosi.UnimplementedProvisionerServer
+}
+
+var _ = Describe("gRPC Factory Client", func() {
+	var (
+		client     *grpcfactory.COSIProvisionerClient
+		grpcServer *grpc.Server
+		listener   net.Listener
+		address    string
+	)
+
+	BeforeEach(func() {
+		address = "unix:///tmp/test.sock"
+		grpcServer = grpc.NewServer()
+
+		// Remove any existing socket file to avoid "address already in use" errors
+		_ = os.Remove(address[7:])
```

Comment:
(slightly nitpicky)

`os.Remove(strings.TrimPrefix(address, "unix://"))` avoids having a magical 7 appear without context.

Reply:
changed, thanks for catching the magic number ;)